modules/dnn_superres/src/dnn_superres.cpp

```diff
@@ -91,6 +91,13 @@ void DnnSuperResImpl::setModel(const String& algo, int scale)
     this->alg = algo;
 }
 
+void DnnSuperResImpl::setCUDA()
+{
+    net.setPreferableBackend(cv::dnn::DNN_BACKEND_CUDA);
+    net.setPreferableTarget(cv::dnn::DNN_TARGET_CUDA);
```

Comment:
More robust solution is to proxy these calls instead:
- setPreferableBackend
- setPreferableTarget

Otherwise we will get soon setOpenCL, setCUDA_FP16, etc.

Reply:
You mean like this? (See new commit)

Reply:
Okay, good. Should I add a note in the tutorials somewhere? And I don't think a test is needed (or possible) for this right?

Reply:
I believe this change is straightforward and doesn't require extra documentation or tests.